yarn-project/prover-node/src/prover-node.ts

```diff
@@ -169,6 +171,7 @@ export class ProverNode implements ClaimsMonitorHandler, EpochMonitorHandler {
     this.publisher.interrupt();
     await Promise.all(Array.from(this.jobs.values()).map(job => job.stop()));
     await this.worldState.stop();
+    await this.p2pClient?.stop();
```

Comment:
If we added a `stop` on the `ProverCoordination` interface, could we avoid passing a `p2pClient` *and* a `coordination` to the constructor and instead just a `coordination`?

Reply:
I did this, originally in https://github.com/AztecProtocol/aztec-packages/pull/9325/commits/cfbbf13c45380360f172d866fbf9e10956bf5b5b but this absolutely fucked up the test instances where we passed in the aztec node, as it called stop on that and killed everything

Reply:
On second thought, if i just restrict the prover client such that it cannot embed an aztec node then this issue will go away